app/transformations/LandOrPropertyTransformer.scala

```diff
@@ -47,19 +48,25 @@ class LandOrPropertyTransformer @Inject() extends Transformer {
   def transformToEtmp(srn: Srn, optLandOrPropertyHeld: Option[Boolean], initialUA: UserAnswers)(
     implicit request: DataRequest[_]
   ): Option[LandOrProperty] =
-    optLandOrPropertyHeld.map(landOrPropertyHeld => {
-      val disposeAnyLandOrProperty = request.userAnswers.get(LandOrPropertyDisposalPage(srn)).getOrElse(false)
+    Option.when(optLandOrPropertyHeld.nonEmpty || request.userAnswers.map(LandPropertyInUKPages(srn)).toList.nonEmpty) {
+      val optDisposeAnyLandOrProperty = request.userAnswers.get(LandOrPropertyDisposalPage(srn))
+      val dispose =
+        if (isPrePopulation(implicitly) && optDisposeAnyLandOrProperty.isEmpty) {
```

Comment:
same here as above.
Should we reuse following method ? This is second time I have seen it.
private def isPrePopulation(implicit request: DataRequest[_]): Boolean =
    request.session.get(PREPOPULATION_FLAG).fold(false)(_.toBoolean)

Reply:
I agree, can I tidy that up in my next ticket @cyeniceri ?